src/PlottingInterface.cc

```diff
@@ -306,7 +437,7 @@ void Transport::Unsubscribe(const std::string &_topic,
 }
 
 ////////////////////////////////////////////
-void Transport::Subscribe(const std::string &_topic,
+Topic *Transport::Subscribe(const std::string &_topic,
```

Comment:
We need to be extra careful when passing raw pointers around, because they can be deleted before someone tries to access them. We should always use smart pointers if we can.

I see that this pointer is being used to set the time reference at the moment of subscription. Would it be possible to hold that time as a `std::shared_ptr` and pass it here as an argument. For example:

```
void Transport::Subscribe(const std::string &_topic,
                          const std::string &_fieldPath,
                          int _chart, const std::shared_ptr<double> &_time)
```

And 

```
void Topic::SetPlottingTimeRef(const std::shared_ptr<double> &_timePtr)
```

And

```
public: std::shared_ptr<double> time;
```

This way, multiple objects can hold pointers to the same time, and none of them can delete the pointer and leave the rest with a null pointer. 

For this function, if you really need to return a double (not a pointer), you could return a copy of the value, something like this:

```
double PlottingInterface::Time() const
{
  return *this->dataPtr->time;
}
```

Reply:
thanks for that :D